columnflow/tasks/plotting.py

```diff
@@ -98,60 +94,92 @@ def run(self):
+                    process_inst: [sub for sub, _, _ in process_inst.walk_processes(include_self=True)]
+                    for process_inst in process_insts
+                }
+                hists[config] = {}
+
+                for dataset, inp in dataset_dict.items():
+                    dataset_inst = config_inst.get_dataset(dataset)
+                    h_in = inp["collection"][0]["hists"].targets[self.branch_data.variable].load(formatter="pickle")
+
+                    # loop and extract one histogram per process
+                    for process_inst in process_insts:
+                        # skip when the dataset is already known to not contain any sub process
+                        if not any(
+                            dataset_inst.has_process(sub_process_inst.name)
+                            for sub_process_inst in sub_process_insts[process_inst]
+                        ):
+                            continue
+
+                        # select processes and reduce axis
+                        h = h_in.copy()
+                        h = h[{
+                            "process": [
+                                hist.loc(p.id)
+                                for p in sub_process_insts[process_inst]
+                                if p.id in h.axes["process"]
+                            ],
+                        }]
+                        h = h[{"process": sum}]
+
+                        # add the histogram
+                        if process_inst in hists[config]:
+                            hists[config][process_inst] += h
+                        else:
+                            hists[config][process_inst] = h
+
+                # there should be hists to plot
+                if not hists:
+                    raise Exception(
+                        "no histograms found to plot; possible reasons:\n"
+                        "  - requested variable requires columns that were missing during histogramming\n"
+                        "  - selected --processes did not match any value on the process axis of the input histogram",
+                    )
 
             # update histograms using custom hooks
             hists = self.invoke_hist_hooks(hists)
 
-            # add new processes to the end of the list
-            for process_inst in hists:
-                if process_inst not in process_insts:
-                    process_insts.append(process_inst)
+            # merge configs
+            if len(self.config_insts) != 1:
+                process_memory = {}
+                merged_hists = {}
+                for config, _hists in hists.items():
+                    for process_inst, h in _hists.items():
+
+                        if process_inst.id in merged_hists:
+                            merged_hists[process_inst.id] += h
+                        else:
+                            merged_hists[process_inst.id] = h
+                            process_memory[process_inst.id] = process_inst
```

Comment:
why is the extra `process_memory` necessary? Wouldn't it be enough to simply set the `process_inst` as key of `merged_hists`? This would also make replacing `hists` trivial